source/kraken/data_manager.h

```diff
@@ -87,7 +90,10 @@ class DataManager {
         data->is_connected_to_rabbitmq = current_data->is_connected_to_rabbitmq.load();
         current_data = std::move(data);
     }
-    boost::shared_ptr<const Data> get_data() const { return current_data; }
+    boost::shared_ptr<const Data> get_data() const {
+        std::lock_guard<std::mutex> lock(*mutex);
```

Comment:
this line is NOT the cause of the crash. I added this because we have a doubt over the race condition this line my introduce...

Reply:
Don't you effectively disable concurrent request handling with this ? 
If data is behind a mutex, two threads cannot read data at the same time, right ?

Reply:
I think you have right.
It can be a case, the access between get_data() and set_data(). 
We have 2 threads for example: The first do `set_data()`, and during the std::move, the second call get_data()... 
It can be unlikely but it is real !

Reply:
Here we could use boost::shared_mutex (std::shared_mutex since c++17) with : 
std::lock_guard<boost::shared_mutex> lock(write); in set_data
std::shared_lock<boost::shared_mutex> lock(write);    in get_data 
For a multiple reader wingle writer configuration

Reply:
another silly question (I am not that familiar with lock/mutex) : isn't you mutex released at the end of the `get_data` function (i.e. immediately) ?

Reply:
The mutex will be release after making a copy of boost::shared_ptr<const Data>, then the copy will be returned